### ISSUE TITLE
Zed crashes when pressing the º key

### ISSUE THREAD

**Issue by felixsanz** [zed-industries/zed#14670]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

On some keyboard layouts (European ones) there is a key above "TAB" and below "ESC". When pressing it... appears the `º` symbol.

![image](https://github.com/user-attachments/assets/ca070e29-4f0f-4def-a0d8-2b4d726dbe4b)

When pressing it.... zed instantly crashes.

```
[2024-07-17T19:19:05+02:00 ERROR zed::reliability] {
  "thread": "main",
  "payload": "capacity overflow",
  "location_data": {
    "file": "library/alloc/src/raw_vec.rs",
    "line": 25
  },
  "backtrace": [
    "core::panicking::panic_fmt",
    "alloc::raw_vec::capacity_overflow",
    "alloc::raw_vec::handle_error",
    "gpui::platform::linux::platform::<impl gpui::platform::keystroke::Keystroke>::from_xkb",
    "<gpui::platform::linux::wayland::client::WaylandClientStatePtr as wayland_client::event_queue::Dispatch<wayland_client::protocol::wl_keyboard::WlKeyboard,()>>::event",
    "wayland_client::event_queue::queue_callback",
    "<core::cell::RefCell<calloop::sources::DispatcherInner<S,F>> as calloop::sources::EventDispatcher<Data>>::process_events",
    "<gpui::platform::linux::wayland::client::WaylandClient as gpui::platform::linux::platform::LinuxClient>::run",
    "gpui::platform::linux::platform::<impl gpui::platform::Platform for P>::run",
    "gpui::app::App::run",
    "zed::main",
    "std::sys_common::backtrace::__rust_begin_short_backtrace",
    "std::rt::lang_start::{{closure}}",
    "std::rt::lang_start_internal",
    "main",
    "__libc_start_call_main",
    "__libc_start_main_impl",
    "_start"
  ],
  "app_version": "0.143.7",
  "release_channel": "Zed",
  "os_name": "Linux Wayland",
  "os_version": "ubuntu 24.04",
  "architecture": "x86_64",
  "panicked_on": 1721236745705,
  "installation_id": "018a6e98-0886-4901-8233-99e9bd7288e3",
  "session_id": "89553613-3eb2-47dd-a044-2ad1c6cff4e0"
}

```

I'm pressing it by mistake all the time 😂 because I changed the keyboard layout after many years using a 60% keyboard, and when I want to press ESC I'm pressing this key instead. Not funny actually 😭

### Environment

Zed: v0.143.7 (Zed)
OS: Linux Wayland ubuntu 24.04
Memory: 15.5 GiB
Architecture: x86_64

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.

Not applicable

**Comment by apricotbucket28** [comment]
FYI this is a duplicate of https://github.com/zed-industries/zed/issues/14417
The crash should has been fixed since 0.144.x, though I still cannot type the character (in case I wanted to).

> I'm pressing it by mistake all the time

I've pressed it more than once accidentally and had Zed crash... :sweat_smile:

**Comment by felixsanz** [comment]
> FYI this is a duplicate of #14417 The crash should has been fixed since 0.144.x, though I still cannot type the character (in case I wanted to).
> 
> > I'm pressing it by mistake all the time
> 
> I've pressed it more than once accidentally and had Zed crash... 😅

oh sorry, tried to search for that symbol. should have search the error message instead. want me to close this issue? hope can be fixed soon, it's really critical to me! haha
